include/aikido/control/SimBarrettFingerPositionCommandExecutor.hpp

```diff
@@ -12,63 +12,68 @@
-  ///        and collision with objects. 
+  ///        and collision with objects.
+  /// \param _collideWith CollisionGroup to check collision with fingers.
   /// \param _collisionOptions Default is (enableContact=false, binaryCheck=true,
-  ///        maxNumContacts = 1.) 
-  ///        See dart/collison/Option.h for more information 
-  BarrettFingerPositionCommandExecutor(
+  ///        maxNumContacts = 1.)
+  ///        See dart/collison/Option.h for more information
+  SimBarrettFingerPositionCommandExecutor(
     ::dart::dynamics::ChainPtr _finger, int _proximal, int _distal,
     ::dart::collision::CollisionDetectorPtr _collisionDetector,
+    ::dart::collision::CollisionGroupPtr _collideWith,
     ::dart::collision::CollisionOption _collisionOptions
       = ::dart::collision::CollisionOption(false, 1));
 
   /// Sets variables to move the finger joints.
   /// proximal dof moves to _goalPosition, joint limit, or until collision.
   /// Step method should be called multiple times until future returns.
-  /// Distal dof follows with mimic ratio. 
+  /// Distal dof follows with mimic ratio.
   /// \param _goalPosition Desired angle of proximal joint.
-  /// \param _collideWith CollisionGroup to check collision with fingers.
   /// \return future Becomes available when the execution completes.
-  std::future<void> execute(double _goalPosition,
-    ::dart::collision::CollisionGroupPtr _collideWith);
+  std::future<void> execute(double _goalPosition);
 
-  /// Returns mimic ratio, i.e. how much the distal joint moves relative to 
+  /// Returns mimic ratio, i.e. how much the distal joint moves relative to
   /// the proximal joint. The joint movements follow
   /// this ratio only when both joints are moving.
   /// \return mimic ratio.
   static double getMimicRatio();
 
   /// Moves the joints of the finger by dofVelocity*_timeSIncePreviousCall
   /// until execute's goalPosition by primary dof or collision is detected.
-  /// If proximal link is in collision, distal link moves until 
+  /// If proximal link is in collision, distal link moves until
   /// mimicRatio*goalPosition. If distal link is in collision, execution stops.
   /// If multiple threads are accessing this function or skeleton associated
   /// with this executor, it is necessary to lock the skeleton before
   /// calling this method.
-  /// \param _timeSincePreviousCall Time since previous call. 
+  /// \param _timeSincePreviousCall Time since previous call.
   void step(double _timeSincePreviousCall);
 
-private: 
-  constexpr static double kMimicRatio = 0.333; 
+  /// Resets CollisionGroup to check collision with fingers.
+  /// \param _collideWith CollisionGroup to check collision with fingers.
+  /// \return false if fails to change collideWith (during execution).
+  bool setCollideWith(::dart::collision::CollisionGroupPtr collideWith);
+
+private:
+  constexpr static double kMimicRatio = 0.333;
   constexpr static double kProximalVelocity = 0.01;
```

Comment:
This is not added by this PR, but renaming `kProximalVelocity` to `kProximalSpeed` would make more sense since the use of this value seems the value doesn't encode the direction of motion.